### ISSUE TITLE
Index boolean assumptions different between sqlite and postgres

### ISSUE THREAD

**Issue by tkinz27** [airbnb/knowledge-repo#203]
Auto-reviewers: @NiharikaRay @matthewwardrop @earthmancash @danfrankj

While trying to run the knowledge repo with postgres as the backing database we kept getting this backtrace.

```python
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/gunicorn/workers/sync.py", line 135, in handle
    self.handle_request(listener, req, client, addr)
  File "/usr/local/lib/python3.5/dist-packages/gunicorn/workers/sync.py", line 176, in handle_request
    respiter = self.wsgi(environ, resp.start_response)
  File "/usr/local/lib/python3.5/dist-packages/flask/app.py", line 1994, in __call__
    return self.wsgi_app(environ, start_response)
  File "/usr/local/lib/python3.5/dist-packages/flask/app.py", line 1985, in wsgi_app
    response = self.handle_exception(e)
  File "/usr/local/lib/python3.5/dist-packages/flask/app.py", line 1547, in handle_exception
    return self.finalize_request(handler(e), from_error_handler=True)
  File "/usr/local/lib/python3.5/dist-packages/knowledge_repo/app/app.py", line 103, in show_traceback
    return render_template('error.html')
  File "/usr/local/lib/python3.5/dist-packages/flask/templating.py", line 132, in render_template
    ctx.app.update_template_context(context)
  File "/usr/local/lib/python3.5/dist-packages/flask/app.py", line 764, in update_template_context
    context.update(func())
  File "/usr/local/lib/python3.5/dist-packages/knowledge_repo/app/app.py", line 153, in inject_version
    last_index=time_since_index(human_readable=True),
  File "/usr/local/lib/python3.5/dist-packages/knowledge_repo/app/index.py", line 27, in time_since_index
    if is_indexing():
  File "/usr/local/lib/python3.5/dist-packages/knowledge_repo/app/index.py", line 20, in is_indexing
    return bool(int(IndexMetadata.get('lock', 'index', '0'))) and last_update is not None and (last_update < timeout)
ValueError: invalid literal for int() with base 10: 'false'
```

We made a quick fix below to get past this issue, but were not sure it was the best solution so we haven't created a pull request directly. 

```diff
diff --git a/knowledge_repo/app/index.py b/knowledge_repo/app/index.py
index 1419fa6..bfd3db4 100644
--- a/knowledge_repo/app/index.py
+++ b/knowledge_repo/app/index.py
@@ -63,7 +63,7 @@ def update_index_required():
                 return True
         return False
     finally:
-        IndexMetadata.set('check', 'index', True)
+        IndexMetadata.set('check', 'index', 1)
         db_session.commit()
@@ -95,7 +95,7 @@ def _update_index(app, force=False, reindex=False):
     if not force and is_indexing():
         return
-    IndexMetadata.set('lock', 'index', True)
+    IndexMetadata.set('lock', 'index', 1)
     db_session.commit()
     kr_dir = {kp.path: kp for kp in current_repo.posts()}
@@ -145,7 +145,7 @@ def _update_index(app, force=False, reindex=False):
     for uri, revision in current_repo.revisions.items():
         IndexMetadata.set('repository_revision', uri, str(revision))
-    IndexMetadata.set('lock', 'index', False)
+    IndexMetadata.set('lock', 'index', 0)
     db_session.commit()
```

**Comment by matthewwardrop** [comment]
@tkinz27 Many apologies for the delay! We had a big company conference a week back, and it set us behind in responding to these issues. Many thanks for passing on these errors to us!

We have in several cases inadvertently used SQL that is idiomatic to MySQL and SQLite, especially where it comes to the equivalent of boolean types with integers. I'll submit a patch with your fixes in the next day or so, and try to get rid of all other potential database issues too.

Again, many thanks!

**Comment by matthewwardrop** [comment]
@tkinz27 This is now fixed by #229.
